docs/developer-notes.md

```diff
@@ -1,8 +1,62 @@
 # Developer notes
 
+## Contributing / Local Warnet Development
+
+### Download the code repository
+
+```bash
+git clone https://github.com/bitcoin-dev-project/warnet
+cd warnet
+```
+
+### Recommended: use a virtual Python environment such as `venv`
+
+```bash
+python3 -m venv .venv # Use alternative venv manager if desired
+source .venv/bin/activate
+```
+
+```bash
+pip install --upgrade pip
+pip install -e .
+```
+
+## Lint
+
 This project primarily uses the `uv` python packaging tool: https://docs.astral.sh/uv/ along with the sister formatter/linter `ruff` https://docs.astral.sh/ruff/
 
 With `uv` installed you can add/remove dependencies using `uv add <dep>` or `uv remove <dep>.
 This will update the [`uv.lock`](https://docs.astral.sh/uv/guides/projects/#uvlock) file automatically.
 
+
 `uv` can also run tools (like `ruff`) without external installation, simply run `uvx ruff check .` or `uvx ruff format .` to use a uv-managed format/lint on the project.
+
+## Release process
```

Comment:
Think this is better in a dedicated document? It's not really developer notes or even relevant for general contributors.

Reply:
only if you insist. i prefer having all the meta stuff in one place whether its for new contributors or us old salts.